InnoGotchiBackend/InnnoGotchi.DAL/Respositories/PictureRepository.cs

```diff
@@ -0,0 +1,69 @@
+﻿using InnnoGotchi.DAL.EF;
+using InnnoGotchi.DAL.Entities;
+using InnnoGotchi.DAL.Interfaces;
+
+namespace InnnoGotchi.DAL.Respositories
+{
+    /// <summary>
+    /// Represents repository that has full access to picture entities in database
+    /// </summary>
+    public class PictureRepository : IRepository<Picture>
+    {
+        private InnoGotchiContext _context;
+
+        public PictureRepository(InnoGotchiContext context)
+        {
+            _context = context;
+        }
+        public bool Contains(Func<Picture, bool> predicate)
+        {
+            Picture? picture = FirstOrDefault(predicate);
+            if (picture == null)
+                return false;
+
+            return true;
+        }
+
+        public void Create(Picture item)
+        {
+            _context.Pictures.Add(item);
+        }
+
+        public bool Delete(int id)
+        {
+            Picture? picture = Get(id);
+            if (picture != null)
+            {
+                _context.Pictures.Remove(picture);
+                return true;
+            }
+            else
+                return false;
+        }
+
+        public IQueryable<Picture> FindAll(Func<Picture, bool> predicate)
+        {
+            return GetAll().Where(predicate).AsQueryable();
```

Comment:
AsQueryable() is really needed here?

Reply:
Method GetAll returns IQueryable but Method Where returns IEnumerable here so I use AsQueryable, or I should in method FindAll return IEnumerable?
![image](https://user-images.githubusercontent.com/74291930/216785092-45bd2cd2-565a-4941-acda-9e75d43b0c75.png)

Reply:
I resolved this problem, now in FindAll parameters insted of Func<Picture, bool> I use Expression<Func<Picture, bool>>. And now method Where returns IQueryable
![image](https://user-images.githubusercontent.com/74291930/216786306-e483c456-cae1-4d6d-b7f3-4114982f42b2.png)